File: MetaBrain/providers/provider-flags.ts
import type { ProviderFeatureFlags } from "./types";

export const DEFAULT_PROVIDER_FLAGS: ProviderFeatureFlags = {
  routerEnabled: false,
  shadowMode: true,
  fallbackEnabled: false,
  healthcheckEnabled: true,
  structuredOutputEnabled: false,
  multimodalEnabled: false,
  externalImageEnabled: false,
  phiAllowed: false,
};

const TRUE_VALUES = new Set(["1", "true", "yes", "on"]);
const FALSE_VALUES = new Set(["0", "false", "no", "off"]);

function readBoolean(value: string | undefined, fallback: boolean): boolean {
  if (value === undefined) return fallback;
  const normalized = value.trim().toLowerCase();
  if (TRUE_VALUES.has(normalized)) return true;
  if (FALSE_VALUES.has(normalized)) return false;
  return fallback;
}

export function loadProviderFlags(env: Record<string, string | undefined> = process.env): ProviderFeatureFlags {
  return {
    routerEnabled: readBoolean(env.LLM_PROVIDER_ROUTER_ENABLED, DEFAULT_PROVIDER_FLAGS.routerEnabled),
    shadowMode: readBoolean(env.LLM_PROVIDER_SHADOW_MODE, DEFAULT_PROVIDER_FLAGS.shadowMode),
    fallbackEnabled: readBoolean(env.LLM_PROVIDER_FALLBACK_ENABLED, DEFAULT_PROVIDER_FLAGS.fallbackEnabled),
    healthcheckEnabled: readBoolean(env.LLM_PROVIDER_HEALTHCHECK_ENABLED, DEFAULT_PROVIDER_FLAGS.healthcheckEnabled),
    structuredOutputEnabled: readBoolean(
      env.LLM_PROVIDER_STRUCTURED_OUTPUT_ENABLED,
      DEFAULT_PROVIDER_FLAGS.structuredOutputEnabled,
    ),
    multimodalEnabled: readBoolean(env.LLM_PROVIDER_MULTIMODAL_ENABLED, DEFAULT_PROVIDER_FLAGS.multimodalEnabled),
    externalImageEnabled: readBoolean(
      env.LLM_PROVIDER_EXTERNAL_IMAGE_ENABLED,
      DEFAULT_PROVIDER_FLAGS.externalImageEnabled,
    ),
    phiAllowed: readBoolean(env.LLM_PROVIDER_PHI_ALLOWED, DEFAULT_PROVIDER_FLAGS.phiAllowed),
  };
}

export const DOCUMENTED_PROVIDER_FLAGS = {
  LLM_PROVIDER_ROUTER_ENABLED: "false",
  LLM_PROVIDER_SHADOW_MODE: "true",
  LLM_PROVIDER_FALLBACK_ENABLED: "false",
  LLM_PROVIDER_HEALTHCHECK_ENABLED: "true",
  LLM_PROVIDER_STRUCTURED_OUTPUT_ENABLED: "false",
  LLM_PROVIDER_MULTIMODAL_ENABLED: "false",
  LLM_PROVIDER_EXTERNAL_IMAGE_ENABLED: "false",
  LLM_PROVIDER_PHI_ALLOWED: "false",
} as const;

File: MB-Secretaria/providers_py/provider_flags.py
from __future__ import annotations

import os
from typing import Optional

from .types import ProviderFlags


def _read_bool(value: Optional[str], fallback: bool) -> bool:
    if value is None:
        return fallback
    normalized = value.strip().lower()
    if normalized in {"1", "true", "yes", "on"}:
        return True
    if normalized in {"0", "false", "no", "off"}:
        return False
    return fallback


def load_provider_flags(env: dict[str, str] | None = None) -> ProviderFlags:
    values = env if env is not None else os.environ
    return ProviderFlags(
        router_enabled=_read_bool(values.get("LLM_PROVIDER_ROUTER_ENABLED"), False),
        shadow_mode=_read_bool(values.get("LLM_PROVIDER_SHADOW_MODE"), True),
        fallback_enabled=_read_bool(values.get("LLM_PROVIDER_FALLBACK_ENABLED"), False),
        healthcheck_enabled=_read_bool(values.get("LLM_PROVIDER_HEALTHCHECK_ENABLED"), True),
        structured_output_enabled=_read_bool(values.get("LLM_PROVIDER_STRUCTURED_OUTPUT_ENABLED"), False),
        multimodal_enabled=_read_bool(values.get("LLM_PROVIDER_MULTIMODAL_ENABLED"), False),
        external_image_enabled=_read_bool(values.get("LLM_PROVIDER_EXTERNAL_IMAGE_ENABLED"), False),
        phi_allowed=_read_bool(values.get("LLM_PROVIDER_PHI_ALLOWED"), False),
    )
